kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ProduceControllerImpl.java

```diff
@@ -32,10 +40,31 @@
 final class ProduceControllerImpl implements ProduceController {
 
   private final Producer<byte[], byte[]> producer;
+  private final KafkaRestConfig config;
```

Comment:
I think we should rate-limit in the resource layer, not in the controller layer. :|